Heart/Character/Domain/Entities/LevelEntity.php

```diff
@@ -69,12 +69,22 @@ public function generateExperience(string $message, bool $isSupporter = false):
         $averageMessageLength = 25;
         $memberStatusMultiplier = $isSupporter ? 0.25 : 0.4;
         $messagePound = ($messageLength / $averageMessageLength);
-        $experienceObtained = ($messagePound / ($this->level * $memberStatusMultiplier) * 20);
+
+        $experienceObtained = ($messagePound / ($this->checkLevelEntity($this->level) * $memberStatusMultiplier) * 20);
         $this->addExperience($experienceObtained);
 
         return (int) $experienceObtained;
     }
 
+    private function checkLevelEntity(int $level): int
+    {
+        if ($this->level !== 0) {
```

Comment:
Uma sugestão para esse if, é mais pra redução de linha/estética:

`return $this->level !== 0  ? $level : 1;`